manageiq-operator/pkg/helpers/miq-components/util.go

```diff
@@ -40,3 +44,107 @@ func addResourceReqs(memLimit string, memReq string, cpuReq string, c *corev1.Co
 
 	return nil
 }
+
+func serviceMutateFn(service *corev1.Service, labels map[string]string, ports []corev1.ServicePort, selector map[string]string) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		service.ObjectMeta.Labels = labels
+		service.Spec.Ports = ports
+		service.Spec.Selector = selector
+		return nil
+	}
+
+	return mutateFn
+}
+
+func secretMutateFn(secret *corev1.Secret, labels map[string]string, data map[string][]byte, stringData map[string]string) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		secret.ObjectMeta.Labels = labels
+		if len(data) > 0 {
+			secret.Data = data
+		}
+		if len(stringData) > 0 {
+			secret.StringData = stringData
+		}
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func configMapMutateFn(configMap *corev1.ConfigMap, labels map[string]string, data map[string]string) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		configMap.ObjectMeta.Labels = labels
+		if len(data) > 0 {
+			configMap.Data = data
+		}
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func pvcMutateFn(pvc *corev1.PersistentVolumeClaim, labels map[string]string, accessModes []corev1.PersistentVolumeAccessMode, resources corev1.ResourceRequirements) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		pvc.ObjectMeta.Labels = labels
+		pvc.Spec.AccessModes = accessModes
+		pvc.Spec.Resources = resources
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func roleMutateFn(role *rbacv1.Role, rules []rbacv1.PolicyRule) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		role.Rules = rules
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func roleBindingMutateFn(roleBinding *rbacv1.RoleBinding, roleRef rbacv1.RoleRef, subjects []rbacv1.Subject) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		roleBinding.RoleRef = roleRef
+		roleBinding.Subjects = subjects
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func deploymentMutateFn(deployment *appsv1.Deployment, labels map[string]string, repNum *int32, container corev1.Container, podTemplateSpec corev1.PodTemplateSpec, podLabels map[string]string) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		deployment.ObjectMeta.Labels = labels
+		deployment.Spec.Strategy.Type = "Recreate"
+		deployment.Spec.Replicas = repNum
+
+		selector := &metav1.LabelSelector{}
+		selector.MatchLabels = podLabels
+		deployment.Spec.Selector = selector
+
+		deployment.Spec.Template = podTemplateSpec
+		deployment.Spec.Template.ObjectMeta.Labels = podLabels
+		deployment.Spec.Template.Spec.Containers = append([]corev1.Container{}, container)
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func NilMutateFn() error {
```

Comment:
I would rather return this from the public service account functions than make this public. It makes them more similar to the rest of the functions that return the openshift objects.